modules/RTC/RTCUtils.js

```diff
@@ -592,6 +592,9 @@ function onMediaDevicesListChanged(devicesReceived) {
 
     sendDeviceListToAnalytics(availableDevices);
 
+    // Used by tracks to update the real device id before the consumer of lib-jitsi-meet receives the new device list.
+    eventEmitter.emit(RTCEvents.DEVICE_LIST_WILL_CHANGE, devicesReceived);
```

Comment:
This event name seems misleading, because the device list has already changed and there is nothing between will change and has changed. But I'm assuming your added it to make sure certain listeners get processed first. Is that correct? Were there any alternatives you were thinking about to intercept DEVICE_LIST_CHANGED to make sure tracks get updated first?

Reply:
Correct! I agree the name may be a little misleading, although you can argue that the list is changed when DEVICE_LIST_CHANGED is fired. Unfortunately I can't think of better name. Any ideas?

Reply:
RECEIVED? That sounds kind of dumb too. At least change the DEVICE_LIST_WILL_CHANGE callback from _onDeviceListChanged to something not clashing with DEVICE_LIST_CHANGED.

Reply:
Changed the name of the method.